### PR TITLE
Docs -> StackNumberAllocator (new file)

### PR DESCRIPTION
**Pull request by paum3**
Documenting an undovumented class. 

## Purpose and Motivation
missing documentation

## Types of changes

- Documentation

## To-do list

- [x] Code is tested
- [x] All tests are passing
- [x] Updated documentation
- [x] This PR is ready for review

### CODE REVIEW

File: HelpSource/Classes/StackNumberAllocator.schelp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+INSTANCEMETHODS::
+
+METHOD:: init
+Reset allocator to its initial state.
+
+METHOD:: alloc
+Allocate next number. After hi value returns nil.
+
+METHOD:: free
```
Comment (cdbzb) [inline HelpSource/Classes/StackNumberAllocator.schelp:27]: lets add a line here:
`argument::inIndex`

File: HelpSource/Classes/StackNumberAllocator.schelp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+Set the next value of the allocator to the arbitrary TELETYPE::inIndex:: (could be any object). Then continue with the counter.
+
+EXAMPLES::
+
+code::
+o = StackNumberAllocator(0,100);
+o.alloc // get the next value, press more times
```
Comment (cdbzb) [inline HelpSource/Classes/StackNumberAllocator.schelp:34]: i think maybe:
`//return the next value (evaluate multiple times)`

File: HelpSource/Classes/StackNumberAllocator.schelp [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+o.init // reset counter
+o.alloc // 0
+
+o.free(2000)// next value (could be any object, eg. String etc.)
```
Comment (cdbzb) [inline HelpSource/Classes/StackNumberAllocator.schelp:39]: let's make this more explicit:
o.free(2000) // returns the StackNumberAllocator itself - sets the next value to 2000 
o.alloc // 2000
o.free("foo") // you can set the next value to any Object, not just integers
o.alloc // "foo"
o.alloc // allocator resumes counting where it left off before we called `.free`

File: HelpSource/Classes/StackNumberAllocator.schelp [hunk 4]
```diff
@@ -0,0 +1,44 @@
+TITLE:: StackNumberAllocator
+summary:: continuous number allocator
```
Comment (cdbzb) [inline HelpSource/Classes/StackNumberAllocator.schelp:2]: i think this should be "consecutive number allocator" !

**Review by cdbzb — COMMENTED** [review]
thanks for this! I wasn't familiar with this class

**Review by cdbzb — APPROVED** [review]

**Review by mossheim — APPROVED** [review]
thanks!

**Pull request merged**